contracts/token/ERC20/ERC20Channel.sol

```diff
@@ -0,0 +1,202 @@
+pragma solidity ^0.4.18;
+
+import "./ERC20.sol";
+import "../../math/SafeMath.sol";
+import "../../ownership/Ownable.sol";
+import "../../ECRecovery.sol";
+
+/**
+   @title ERC20Channel, State channel for ERC20 Tokens
+
+   Contract that provides holders of a ERC20 compatible token the creation of a
+   state channel between two users, once a channel is open the users can exchange
+   signatures offchain to agree on the final value of the transfer.
+   Uses OpenZeppelin ERC20 and SafeMath lib.
+
+   Note: The owner of the contract is the sender, therefore it should be
+    deployed by the sender itself.
+
+   The channel can be closed in two ways: With an agreement or not.
+
+   - Contract closed with an agreement:
+     Can happen at anytime while the channel is still opened. Two signatures
+     are needed for this, one is the signature of the receiver agreeing to
+     receive a certain value, with the receiver signature the sender can now
+     agree on the value and use sign the receiver signature.
+
+      cooperativeClose(
+       FinalBalance,
+       Receiver.sign(FinalBalance),
+       Sender.sign(SHA3( Receiver.sign(FinalBalance) ))
+      )
+
+   - Contract closed without agreement:
+    This can happen only in behalf of the sender, and it has a time to be
+     "challenged" by the receiver. The sender and receiver exchange signatures
+     of a final value to be transfered but they reach a point where they dont
+     agree on the final value. The sender can ask to close the channel with a
+     final value, if the challenge time passes and the channel does not receive
+     a cooperativeClose request it would be able to be closed by the sender
+     transfering the final value requested.
+ */
+contract ERC20Channel is Ownable {
```

Comment:
I've just realized that a sender can at any time `transferOwnership` of the contract, changing the sender and invalidating all previous signed messages,  since `getSignerOfBalanceHash` must be equal to the contract owner. So a malicious sender could transfer ownership of this contract to a different address he controls (after having sent several correctly signed payment messages with the original address), and run an uncooperative close to retrieve his funds. We'll need to remove the Ownable inheritance.